src/main/java/com/basho/riak/client/core/converters/TimeSeriesConverter.java

```diff
@@ -0,0 +1,203 @@
+package com.basho.riak.client.core.converters;
+
+import com.basho.riak.client.core.query.timeseries.Cell;
+import com.basho.riak.client.core.query.timeseries.ColumnDescription;
+import com.basho.riak.client.core.query.timeseries.QueryResult;
+import com.basho.riak.client.core.query.timeseries.Row;
+import com.basho.riak.client.core.util.BinaryValue;
+import com.basho.riak.protobuf.RiakKvPB;
+import com.google.protobuf.ByteString;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * @author Alex Moore <amoore at basho dot com>
+ * @since 2.0.3
+ */
+public class TimeSeriesConverter
+{
+    private List<Row> parseRows(List<RiakKvPB.TsRow> pbRows)
+    {
+        ArrayList<Row> rows = new ArrayList<Row>();
+
+        for (RiakKvPB.TsRow pbRow : pbRows)
+        {
+            List<Cell> cells = new ArrayList<Cell>();
+
+            for (RiakKvPB.TsCell pbCell : pbRow.getCellsList())
+            {
+                if (pbCell.hasBinaryValue())
+                {
+                    cells.add(new Cell(BinaryValue.unsafeCreate(pbCell.getBinaryValue().toByteArray())));
```

Comment:
You are totally right about this case, but at the moment, since we have the only one protocol from my point of view it's better to introduce template method. Later, if that become a case, we may do simple refactoring to introduce AbstractFactory with PBCellConcreteFactory and WhateEverProtoConcreteFactory..

if we are going to move magic inside the factory method, better to call the method kinda fromPB(..)